downloadClient: Cannot cast object 'true' with class 'Boolean' to class 'Property'

I am trying to enable client downloading as per [docs](https://unbroken-dome.github.io/projects/gradle-helm-plugin/#_automatic_download_of_the_helm_client):
```groovy
// build.gradle
helm {
    downloadClient {
        enabled = true
        version = '3.2.0'
    }
}
```
However, this is what I get on Gradle 6.6:
```
Caused by: org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'true' with class 'java.lang.Boolean' to class 'org.gradle.api.provider.Property'
```
However, Kotlin syntax works:
```groovy
helm {
    downloadClient {
        enabled.set(true)
        version.set("3.2.0")
    }
}
```

It is strange because it seems other closures don't have this problem:
```groovy
helm {
    downloadClient {
        enabled.set(true)
        version.set("3.2.0")
    }
    // works as expected
    lint {
        strict = true
    }
}
```

--- a/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/dsl/HelmDownloadClient.kt
+++ b/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/dsl/HelmDownloadClient.kt
@@ -12,6 +12,7 @@ import org.unbrokendome.gradle.plugins.helm.util.GRADLE_VERSION_6_2
 import org.unbrokendome.gradle.plugins.helm.util.booleanProviderFromProjectProperty
 import org.unbrokendome.gradle.plugins.helm.util.property
 import org.unbrokendome.gradle.plugins.helm.util.providerFromProjectProperty
+import javax.inject.Inject
 
 
 /**
@@ -72,8 +73,8 @@ internal interface HelmDownloadClientInternal : HelmDownloadClient {
 }
 
 
-internal class DefaultHelmDownloadClient
-constructor(
+internal open class DefaultHelmDownloadClient
+@Inject constructor(
     private val project: Project
 ) : HelmDownloadClient, HelmDownloadClientInternal {
 

--- a/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/dsl/HelmExtension.kt
+++ b/src/main/kotlin/org/unbrokendome/gradle/plugins/helm/dsl/HelmExtension.kt
@@ -72,7 +72,7 @@ private open class DefaultHelmExtension
     ConfigurableHelmServerOptions by HelmServerOptionsHolder(objects).applyConventions(project) {
 
     final override val downloadClient: HelmDownloadClient =
-        DefaultHelmDownloadClient(project)
+        objects.newInstance(DefaultHelmDownloadClient::class.java, project)
 
 
     final override val executable: Property<String> =

